gpflow/config/__config__.py

```diff
@@ -1,3 +1,40 @@
+"""
+This is a private module that manages GPflow configuration.
+
+The module provides functions to modify default settings of GPflow, such as:
+the standard float precision and integer type, the positive transformation
+type, a value for a minimum shift from zero for the positive transformation,
+an output format for the summary printing.
+
+The module holds global configuration :class:`Config` variable that stores all
+setting values.
+
+The environment variables is an alternative way for changing default GPflow
+configuration.
+
+.. warning::
+    The user has to set environment variables before running python
+    interpreter to modify the configuration.
+
+Full set of environment variables and available options:
+    ```
+    GPFLOW_INT: "int16", "int32", or "int64"
+    GPFLOW_FLOAT: "float16", "float32", or "float64"
+    GPFLOW_POSITIVE_BIJECTOR: "exp" or "softplus"
+    GPFLOW_POSITIVE_MINIMUM: Any positive float number
+    GPFLOW_SUMMARY_FMT: "notebook" or any other format that
+        :mod:`tabulate` can handle.
+    GPFLOW_JITTER: Any positive float number
+    ```
+
+The user also can change GPflow configuration temporarily with a context
+manager :func:`as_context`:
+
+>>> config = Config(jitter=1e-5)
```

Comment:
I don't see any examples of this in the notebooks - should we add an example usage?

Reply:
yes, would be great to add this to one of the intro notebooks (intro_to_gpflow2 maybe)?

Reply:
I will address it in a separate PR.